api/proto/teleport/legacy/types/types.proto

```diff
@@ -6835,6 +6835,23 @@ message PluginAWSICSettings {
   // CredentialsSource indicates how the Identity Center plugin should source
   // its AWS login credentials
   AWSICCredentialsSource credentials_source = 7;
+
+  // UserSyncLabelsFilter specifies a map of key-value pairs used to filter users
+  // based on their metadata labels. These filtered users will be provisioned
+  // from Teleport to AWS IC via SCIM provisioning.
+  //
+  // Example:
+  // If Okta is still used as the Identity Source and only users originating from Okta
+  // should be synced, set the filter to:
+  // {
+  //   "okta/org": "https://trial-123456.okta.com",
+  //   "teleport.dev/origin": "okta"
+  // }
+  //
+  // If AWS IC uses Teleport as the Identity Provider, the filter should remain empty.
+  //
+  // NOTE: System users are always filtered out by default and will not be provisioned to AWS IC.
+  map<string, string> user_sync_labels_filter = 8 [(gogoproto.jsontag) = "user_sync_labels_filter,omitempty"];
```

Comment:
Should we allow for multiple matching values for any particular label? e.g. 

```
teleport.dev/origin: okta, aws-identity-center`
```

It's not necessary for this particular deployment, but I can see this coming back to bite us at some point in the future.